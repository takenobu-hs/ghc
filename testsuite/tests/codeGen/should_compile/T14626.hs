{-# LANGUAGE MagicHash #-}

module T14626 where

import GHC.Prim

data T = MkT !Bool

f v = case v of
         MkT y -> dataToTag# y

-- This should /not/ produce an inner case on the y, thus:
--    f v = case v of
--            MkT y -> case y of z -> dataToTag# z
-- But it was!  See Trac #14626 comment:4
