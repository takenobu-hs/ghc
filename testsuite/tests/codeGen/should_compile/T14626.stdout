      case dt of dt { __DEFAULT -> T14626.MkT dt }
      case v of { T14626.MkT y [Occ=Once] ->
